### PR TITLE
sql server user store implementation

### PR DESCRIPTION
**Pull request by kpaul**
still might need to adjust the entities a bit for FK's but the store implementation is done

---
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/91)

### CODE REVIEW

File: Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs [hunk 1]
```diff
@@ -35,12 +35,12 @@ public CouchDbUserStore(IDocumentDbService documentDbService, ILogger logger)
             return user?.FirstOrDefault();
         }
 
-        public Task<IEnumerable<User>> GetUsersBySubjectId(IEnumerable<string> subjectIds)
+        public Task<IEnumerable<User>> GetUsersBySubjectIdAsync(IEnumerable<string> subjectIds)
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs]: Any reason these are not marked async, and we aren't using the await keyword?

Comment (michaelvidal24) [inline Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs]: Any reason this is not marked async, and we are not awaiting in the method? Is the _documentDbService.GetDocumentsById function not async?

File: Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs [hunk 2]
```diff
@@ -35,12 +35,12 @@ public CouchDbUserStore(IDocumentDbService documentDbService, ILogger logger)
...
         {
             return _documentDbService.GetDocumentsById<User>(subjectIds);
         }
 
-        public Task<User> AddUser(User user)
+        public Task<User> AddUserAsync(User user)
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs]: Same comment as above.

File: Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs (hunk 1 above)
Comment (kpaul) [reply Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs]: i just updated the method name and did not check this file to ensure it was marked async.

File: Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs (hunk 2 above)
Comment (kpaul) [reply Fabric.Identity.API/Persistence/CouchDb/Stores/CouchDbUserStore.cs]: same comment as above.

**Pull request closed without merging**